docs/templates/solc_0.8/permit/Permit.md

```diff
@@ -0,0 +1,92 @@
+---
+description: Permit
+---
+
+# Permit
+
+## Introduction
+
+*The purpose of this contract is to allow transfer from an owner ERC-20 extended to a spender one*
+
+## Model
+
+*This contract is dealing with sand (an ERC20 extended).*  
+*[ERC20](https://ethereum.org/en/developers/docs/standards/tokens/erc-20/) is the classic standard to represent fungible token.*  
+*ERC20 extended only had the ability to burn (destroy) tokens.*  
+
+## Process
+
+### Step 1
+
+*we ensure that deadline authorization has not expired*  
+
+### Step 2
+
+*we build the digest message that was signed with owner private key*  
+
+### Step 3
+
+*In elliptic cryptographie ECDSA (Elliptic Curve Digital Signature Algorithm)  
+message signatures are made with two integers: r and s (32 bytes).  
+[Ethereum](https://medium.com/mycrypto/the-magic-of-digital-signatures-on-ethereum-98fe184dc9c7) has added an additional recovery identifier variable called v.
+So the signature computed is made of (v, r, s).*  
+
+*[ecrecover](https://soliditydeveloper.com/ecrecover) is a function that allow us to [recover adress](https://crypto.stackexchange.com/questions/18105/how-does-recovering-the-public-key-from-an-ecdsa-signature-work) ( derived from a public key )  
+associated to the private key that signed a message.  
+we must have both signature (v, r, s) and the original message that was signed with private key  
+in our case the message is digest.*  
+
+*v can either be 27 or 28.  
+On elliptic curve multiple points can be computed from from (r, s) alone.  
+This would result in multiple public key computed from signed message  
+This is why we must have recovery identifier to select the right point and then the right adress.*  
+
+### Step 4
+
+*Once we recovered adress from signature and digest we must check  
+that it is the owner adress and exit overwise.*  
```

Comment:
exit => reject/revert is a more accurate word

Reply:
i agree